python/ray/serve/_private/controller.py

```diff
@@ -115,15 +117,33 @@ async def __init__(
         controller_name: str,
         *,
         http_config: HTTPOptions,
+        logging_config: LoggingConfig,
         grpc_options: Optional[gRPCOptions] = None,
     ):
         self._controller_node_id = ray.get_runtime_context().get_node_id()
         assert (
             self._controller_node_id == get_head_node_id()
         ), "Controller must be on the head node."
 
+        # Try to read config from checkpoint
+        self.ray_worker_namespace = ray.get_runtime_context().namespace
+        self.controller_name = controller_name
+        self.gcs_client = GcsClient(address=ray.get_runtime_context().gcs_address)
+        kv_store_namespace = f"{self.controller_name}-{self.ray_worker_namespace}"
+        self.kv_store = RayInternalKVStore(kv_store_namespace, self.gcs_client)
+        log_config_checkpoint = self.kv_store.get(LOGGING_CONFIG_CHECKPOINT_KEY)
+        if log_config_checkpoint is not None:
+            logging_config = pickle.loads(log_config_checkpoint)
+        self.logging_config: LoggingConfig = logging_config
```

Comment:
Finding this logic hard to follow because the initial config and reconfigured configs are going down different paths (the former is an init arg, the latter goes through `reconfigure`) -- two possible suggestions for cleaning it up:

1. Don't pass the logging config as an init arg at all, and instead after creating the controller immediately call the `reconfigure_system_logging_config` method.
2. Add a helper method for getting the current logging config in the constructor. First check if there is a checkpointed logging config already. If so, use it. If not, call `reconfigure_system_logging_config` directly from the constructor.

I think either of these would be a good unification, lmk what you think.

Reply:
done!